src/elevenlabs/conversation.py

```diff
@@ -0,0 +1,282 @@
+        pass
+
+    @abstractmethod
+    def stop(self):
+        """Stops the audio interface.
+
+        Called one time after the conversation ends. Should clean up any resources
+        used by the audio interface and stop any audio streams. Do not call the
+        `input_callback` from `start` after this method is called.
+        """
+        pass
+
+    @abstractmethod
+    def output(self, audio: bytes):
+        """Output audio to the user.
+
+        The `audio` input is in 16-bit PCM mono format at 16kHz. Implementations can
+        choose to do additional buffering. This method should return quickly and not
+        block the calling thread.
+        """
+        pass
+
+    @abstractmethod
+    def interrupt(self):
+        """Interruption signal to stop any audio output.
+
+        User has interrupted the agent and all previosly buffered audio output should
+        be stopped.
+        """
+        pass
+
+
+class Conversation:
+    client: BaseElevenLabs
+    agent_id: str
+    requires_auth: bool
+
+    audio_interface: AudioInterface
+    callback_agent_response: typing.Optional[typing.Callable[[str], None]]
+    callback_user_transcript: typing.Optional[typing.Callable[[str], None]]
+    callback_latency_measurement: typing.Optional[typing.Callable[[int], None]]
+
+    _thread: typing.Optional[threading.Thread] = None
+    _should_stop: threading.Event = threading.Event()
+    _conversation_id: typing.Optional[str] = None
+    _last_interrupt_id: int = 0
+
+    def __init__(
+        self,
+        client: BaseElevenLabs,
+        agent_id: str,
+        *,
+        requires_auth: bool,
+        audio_interface: AudioInterface,
+        callback_agent_response: typing.Optional[typing.Callable[[str], None]] = None,
+        callback_user_transcript: typing.Optional[typing.Callable[[str], None]] = None,
+        callback_latency_measurement: typing.Optional[typing.Callable[[int], None]] = None,
+    ):
+        """Conversational AI session.
+
+        BETA: This API is subject to change without regard to backwards compatibility.
+
+        Args:
+            client: The ElevenLabs client to use for the conversation.
+            agent_id: The ID of the agent to converse with.
+            requires_auth: Whether the agent requires authentication.
+            audio_interface: The audio interface to use for input and output.
+            callback_agent_response: Callback for agent responses.
+            callback_user_transcript: Callback for user transcripts.
+            callback_latency_measurement: Callback for latency measurements (in milliseconds).
+        """
+
+        self.client = client
+        self.agent_id = agent_id
+        self.requires_auth = requires_auth
+
+        self.audio_interface = audio_interface
+        self.callback_agent_response = callback_agent_response
+        self.callback_user_transcript = callback_user_transcript
+        self.callback_latency_measurement = callback_latency_measurement
```

Comment:
Since most are passed by name, what about pydantic BaseModel? Or you think too restrictive?

Reply:
I considered it but we don't have pydantic as dependency now and so I don't want to add it just to save myself few lines. Any dependency we take is additional constraint on the users of this package.